### PR TITLE
Add new ButtonLink component

### LINKED ISSUE #181: "url" prop for TextLink needs to be optional
Most of the time, all login is handled by `onClick` in react apps and url isn't relevant

### PR DESCRIPTION
**Pull request by wEprous**
This Pull Request meets the following criteria:

* [x] Tests have been added/adjusted for my new feature
* [x] New Components are registered in index.js of my project

closes #181

### CODE REVIEW

File: src/ButtonLink/index.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
+// @flow
+import * as React from "react";
+import defaultTokens from "@kiwicom/orbit-design-tokens";
+import styled from "styled-components";
+import { withTheme } from "theming";
+
+import { iconSizes } from "../Icon";
+
+type Props = {
+  children?: React.Node,
```
Comment (jaroslav-kubicek) [inline src/ButtonLink/index.js]: if the children is optional, I would add such example into stories

Comment (wEprous) [reply src/ButtonLink/index.js]: OK, good point

File: src/ButtonLink/index.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    theme,
+    type,
+    icon,
+    sizeIcon,
+    width,
+    disabled,
+    url,
+    children,
+    ...props
+  }) => <a {...props}>{children}</a>,
```
[the remark this answers is not in the document]
Comment (tomashapl) [reply src/ButtonLink/index.js]: @darkwindcz We make decision tomorrow right ?

[the remark this answers is not in the document]
Comment (darkwindcz) [reply src/ButtonLink/index.js]: yes, I'll try to find something in our calendars but it looks already pretty busy day :/

File: src/ButtonLink/ButtonLink.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  disabled: boolean,
+  block: boolean,
+  external: boolean,
+  type: $Values<typeof TYPES>,
+  size: $Values<typeof SIZES>,
+  href?: string,
+  width: number,
+  icon?: React.Node,
+  theme: typeof defaultTokens,
+  transparent?: boolean,
```
Comment (wEprous) [inline src/ButtonLink/ButtonLink.js]: We can define `transparent` as required and set default to `false`

File: src/Alert/Alert.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+const Content = styled.div`
+  display: flex;
+  align-items: center;
+  min-height: ${({ theme, title }) => !title && theme.heightIconMedium};
+  margin-bottom: ${({ theme, title }) => title && theme.spaceXXSmall};
+`;
+
+const CloseContainer = styled.div`
+  position: absolute;
```
Comment (wEprous) [inline src/Alert/Alert.js]: We can use here `flex` as well, right? Personally, I don't like mixing flex with `position`

Comment (tomashapl) [reply src/Alert/Alert.js]: I don't want to affect layout by close icon, I think `absolute` is better

File: src/ButtonLink/ButtonLink.js (hunk 3 above)
Comment (tomashapl) [reply src/ButtonLink/ButtonLink.js]: Yes, but  i don't see any benefits, the same we can do with other props, but why ?

Comment (wEprous) [reply src/ButtonLink/ButtonLink.js]: Currently, you can see `transparent={false}` in the preview source code.
Also, if someone will copy this code, there is no need to pass `anyProp={false}` to a component.

Comment (wEprous) [reply src/ButtonLink/ButtonLink.js]: It'll probably results in warning/error I think.

File: src/Alert/Alert.js (hunk 4 above)
Comment (wEprous) [reply src/Alert/Alert.js]: OK, as you think, but I rather use only `flex`.
On the **line 90** you have hardcoded `-16px`, it needs to be token.

**Pull request merged**